rpc/lib/server/http_server.go

```diff
@@ -111,6 +111,23 @@ func WriteRPCResponseHTTP(w http.ResponseWriter, res types.RPCResponse) {
 	w.Write(jsonBytes) // nolint: errcheck, gas
 }
 
+// WriteRPCResponseArrayHTTP will do the same as WriteRPCResponseHTTP, except it
+// can write arrays of responses for batched request/response interactions via
+// the JSON RPC.
+func WriteRPCResponseArrayHTTP(w http.ResponseWriter, res []types.RPCResponse) {
+	if len(res) == 1 {
+		WriteRPCResponseHTTP(w, res[0])
+	} else {
+		jsonBytes, err := json.MarshalIndent(res, "", "  ")
+		if err != nil {
+			panic(err)
```

Comment:
maybe return error?

Reply:
There's code in [`rpc/lib/server/http_server.go`](https://github.com/tendermint/tendermint/blob/8dd2798cfcc2fd57ed5c90132f1bfaa4c81d932a/rpc/lib/server/http_server.go#L148) (not my code) that catches panics and returns 500 Internal Server errors.

Should we refactor all of that?

Reply:
no, I think that's fine. But can you please add

```
err := w.Write(jsonBytes)
if err != nil {
  panic(err)
}
```

below and in WriteRPCResponseHTTP function and remove `// nolint: errcheck`?

Reply:
Done 👍